xds/type/matcher/v3/matcher.proto

```diff
@@ -125,18 +125,16 @@ message Matcher {
   }
 
   oneof matcher_type {
-    option (validate.required) = true;
-
     // A linear list of matchers to evaluate.
     MatcherList matcher_list = 1;
 
     // A match tree to evaluate.
     MatcherTree matcher_tree = 2;
   }
 
-  // Optional OnMatch to use if the matcher failed.
-  // If specified, the OnMatch is used, and the matcher is considered
-  // to have matched.
-  // If not specified, the matcher is considered not to have matched.
+  // Optional OnMatch to use if the matcher above was unspecified or if matching using the specified
+  // matcher failed.
+  // If not specified, the match will be considered unsucessful if a matcher was not specified or
+  // if the specified matcher failed to match.
   OnMatch on_no_match = 3;
```

Comment:
The name of this field is a bit of a misnomer now (or maybe not if you wrap your head around an unspecified matcher as meaning "match nothing"), but not sure if we want to rename